src/epd2in13_v2/mod.rs

```diff
@@ -0,0 +1,369 @@
+    for EPD2in13<SPI, CS, BUSY, DC, RST>
+where
+    SPI: Write<u8>,
+    CS: OutputPin,
+    BUSY: InputPin,
+    DC: OutputPin,
+    RST: OutputPin,
+{
+    fn init<DELAY: DelayMs<u8>>(
+        &mut self,
+        spi: &mut SPI,
+        delay: &mut DELAY,
+    ) -> Result<(), SPI::Error> {
+        // Reset the device
+        self.interface.reset(delay);
+
+        self.command(spi, Command::SW_RESET)?;
+
+        self.cmd_with_data(spi, Command::SET_ANALOG_BLOCK_CONTROL, &[0x54])?;
+        self.cmd_with_data(spi, Command::SET_DIGITAL_BLOCK_CONTROL, &[0x3B])?;
+
+        self.cmd_with_data(spi, Command::DRIVER_OUTPUT_CONTROL, &[(HEIGHT-1) as u8, 0x00, 0x00])?;
+
+        self.set_data_entry_mode(spi, DataEntryModeIncr::X_INCR_Y_INCR, DataEntryModeDir::X_DIR)?;
+
+        // start from Gate 0
+        self.cmd_with_data(spi, Command::GATE_SCAN_START_POSITION, &[0x00])?;
+
+//        self.use_full_frame(spi)?;
+
+        self.cmd_with_data(spi, Command::BORDER_WAVEFORM_CONTROL, &[0x03])?;
+
+        // GxEPD:
+        self.cmd_with_data(spi, Command::WRITE_VCOM_REGISTER, &[0x70])?;
+
+        self.cmd_with_data(spi, Command::GATE_DRIVING_VOLTAGE_CTRL, &[0x15])?;
+        self.cmd_with_data(spi, Command::SOURCE_DRIVING_VOLTAGE_CTRL, &[0x41, 0xA8, 0x32])?;
+
+        // default values [POR]
+        self.cmd_with_data(spi, Command::SET_DUMMY_LINE_PERIOD, &[0x30])?;
+        self.cmd_with_data(spi, Command::SET_GATE_LINE_WIDTH, &[0x0A])?;
+
+        self.set_lut(spi, None)?;
+
+        self.wait_until_idle();
+        Ok(())
+    }
+}
+
+impl<SPI, CS, BUSY, DC, RST> WaveshareDisplay<SPI, CS, BUSY, DC, RST>
+    for EPD2in13<SPI, CS, BUSY, DC, RST>
+where
+    SPI: Write<u8>,
+    CS: OutputPin,
+    BUSY: InputPin,
+    DC: OutputPin,
+    RST: OutputPin,
+{
+    fn new<DELAY: DelayMs<u8>>(
+        spi: &mut SPI,
+        cs: CS,
+        busy: BUSY,
+        dc: DC,
+        rst: RST,
+        delay: &mut DELAY,
+    ) -> Result<Self, SPI::Error> {
+        let interface = DisplayInterface::new(cs, busy, dc, rst);
+        let color = DEFAULT_BACKGROUND_COLOR;
+
+        let mut epd = EPD2in13 {
+            interface,
+            background_color: DEFAULT_BACKGROUND_COLOR,
+            refresh: RefreshLUT::FULL,
+        };
+
+        epd.init(spi, delay)?;
+
+        Ok(epd)
+    }
+
+    fn wake_up<DELAY: DelayMs<u8>>(
+        &mut self,
+        spi: &mut SPI,
+        delay: &mut DELAY,
+    ) -> Result<(), SPI::Error> {
+        self.init(spi, delay)
+    }
+
+    fn sleep(&mut self, spi: &mut SPI) -> Result<(), SPI::Error> {
+        self.wait_until_idle();
+
+        // was 0xC3 but nothing in doc.
+        self.cmd_with_data(spi, Command::DISPLAY_UPDATE_CONTROL_2, &[0x03])?;
+        self.command(spi, Command::MASTER_ACTIVATION)?;
+
+        // 00 : normal
+        // 01 : deep sleep mode 1
+        // 11 : deep sleep mode 2
+        self.cmd_with_data(spi, Command::DEEP_SLEEP_MODE, &[0x01])?;
```

Comment:
What is the difference between these sleep modes?

Reply:
No clue yet, as I have not tried to understand this part yet :D

Reply:
I've updated this part. There are 3 modes: 
- 1 where access through MCU is alive including reading RAM and RAM content is kept. 
- 1 where MCU/RAM access is not possible but RAM content is still kept
- 1 where no access is possible and RAM content is not kept.

I've added corresponding enum

Reply:
I think the controller are at least very close, but they have some subtle diff that makes them incompatible. At least where I've looked. It sure would be a good think to factor what can be factored... We would need to have the exact controller used for all displays and see how much they can share.
I know this one is close to 2in9 as command number are the same... But usually, sample code for init is different.